src/pages/stacks-blockchain/integrate-stacking.md

```diff
@@ -0,0 +1,366 @@
+  Configuration,
+  TransactionsApi,
+} = require('@stacks/blockchain-api-client');
+const c32 = require('c32check');
+
+const apiConfig = new Configuration({
+  fetchApi: fetch,
+  basePath: 'https://stacks-node-api.blockstack.org',
+});
+
+// generate rnadom key
+const privateKey = makeRandomPrivKey();
+
+// get Stacks address
+const stxAddress = getAddressFromPrivateKey(
+  privateKeyToString(privateKey),
+  TransactionVersion.Testnet
+);
+```
+
+-> Review the [accounts guide](/stacks-blockchain/accounts) for more details
+
+## Step 3: Display stacking info
+
+In order to inform users about the upcoming reward cycle, we need to obtain Stacking information:
+
+```js
+const info = new InfoApi(apiConfig);
+
+const poxInfo = await info.getPoxInfo();
+const coreInfo = await info.getCoreApiInfo();
+const blocktimeInfo = await info.getNetworkBlockTimes();
+
+console.log({ poxInfo, coreInfo, blocktimeInfo });
+```
+
+-> Check out the API references for the 3 endpoints used here: [GET /v2/info](https://blockstack.github.io/stacks-blockchain-api/#operation/get_core_api_info), [GET v2/pox](https://blockstack.github.io/stacks-blockchain-api/#operation/get_pox_info), and [GET /extended/v1/info/network_block_times](https://blockstack.github.io/stacks-blockchain-api/#operation/get_network_block_times)
+
+-> Stacking execution will differ between mainnet and testnet in terms of cycle times and participation thresholds
+
+With the obtained PoX info, you can present if Stacking is executed in the next cycle, when the next cycle begins, the duration of a cycle, and the minimum micro-Stacks required to participate:
+
+```js
+// will Stacking be executed in the next cycle?
+const stackingExecution = poxInfo.rejection_votes_left_required > 0;
+
+// how long (in seconds) is a Stacking cycle?
+const cycleDuration = poxInfo.reward_cycle_length * blocktimeInfo.testnet.target_block_time;
+
+// how much time is left (in seconds) until the next cycle begins?
+const secondsToNextCycle =
+  (poxInfo.reward_cycle_length -
+    ((coreInfo.burn_block_height - poxInfo.first_burnchain_block_height) %
+      poxInfo.reward_cycle_length)) *
+  blocktimeInfo.testnet.target_block_time;
+
+// the actual datetime of the next cycle start
+const nextCycleStartingAt = new Date();
+nextCycleStartingAt.setSeconds(nextCycleStartingAt.getSeconds() + secondsToNextCycle);
+
+console.log({
+  stackingExecution,
+  cycleDuration,
+  nextCycleStartingAt,
+  // mimnimum micro-Stacks required to participate
+  minimumUSTX: poxInfo.min_amount_ustx,
+});
+```
+
+Users need to have sufficient Stacks (STX) tokens to participate in Stacking. With the Stacking info, this can be verified easily:
+
+```js
+const accounts = new AccountsApi(apiConfig);
+
+const accountBalance = await accounts.getAccountBalance({
+  principal: stxAddress,
+});
+
+const accountSTXBalance = accountBalance.stx.balance;
+
+// enough balance for participation?
+const canParticipate = accountSTXBalance >= poxInfo.min_amount_ustx;
```

Comment:
Should bn.js be used here? AccountSTXBalance could be a very large number

Reply:
yeah, let me add that!